cmd/axelard/cmd/vald/start.go

```diff
@@ -195,87 +195,126 @@ func listen(ctx sdkClient.Context, txf tx.Factory, axelarCfg config.ValdConfig,
 		return tmEvents.MustSubscribeWithAttributes(eventBus,
 			eventType, module, sdk.Attribute{Key: sdk.AttributeKeyAction, Value: action})
 	}
 
+	blockHeaderSub := tmEvents.MustSubscribeBlockHeader(eventBus)
+	subscriptions = append(subscriptions, blockHeaderSub)
+
 	queryHeartBeat := createNewBlockEventQuery(tssTypes.EventTypeHeartBeat, tssTypes.ModuleName, tssTypes.AttributeValueSend)
 	heartbeat, err := tmEvents.Subscribe(eventBus, queryHeartBeat)
 	if err != nil {
 		panic(fmt.Errorf("unable to subscribe with ack event query: %v", err))
 	}
+	subscriptions = append(subscriptions, heartbeat)
 
 	keygenStart := subscribe(tssTypes.EventTypeKeygen, tssTypes.ModuleName, tssTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, keygenStart)
 
 	querySign := createNewBlockEventQuery(tssTypes.EventTypeSign, tssTypes.ModuleName, tssTypes.AttributeValueStart)
 	signStart, err := tmEvents.Subscribe(eventBus, querySign)
 	if err != nil {
 		panic(fmt.Errorf("unable to subscribe with sign event query: %v", err))
 	}
+	subscriptions = append(subscriptions, signStart)
 
 	keygenMsg := subscribe(tssTypes.EventTypeKeygen, tssTypes.ModuleName, tssTypes.AttributeValueMsg)
+	subscriptions = append(subscriptions, keygenMsg)
 	signMsg := subscribe(tssTypes.EventTypeSign, tssTypes.ModuleName, tssTypes.AttributeValueMsg)
+	subscriptions = append(subscriptions, signMsg)
 
 	evmNewChain := subscribe(evmTypes.EventTypeNewChain, evmTypes.ModuleName, evmTypes.AttributeValueUpdate)
+	subscriptions = append(subscriptions, evmNewChain)
 	evmChainConf := subscribe(evmTypes.EventTypeChainConfirmation, evmTypes.ModuleName, evmTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, evmNewChain)
 	evmGatewayDeploymentConf := subscribe(evmTypes.EventTypeGatewayDeploymentConfirmation, evmTypes.ModuleName, evmTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, evmGatewayDeploymentConf)
 	evmDepConf := subscribe(evmTypes.EventTypeDepositConfirmation, evmTypes.ModuleName, evmTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, evmDepConf)
 	evmTokConf := subscribe(evmTypes.EventTypeTokenConfirmation, evmTypes.ModuleName, evmTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, evmTokConf)
 	evmTraConf := subscribe(evmTypes.EventTypeTransferKeyConfirmation, evmTypes.ModuleName, evmTypes.AttributeValueStart)
+	subscriptions = append(subscriptions, evmTraConf)
 
 	eventCtx, cancelEventCtx := context.WithCancel(context.Background())
 	// stop the jobs if process gets interrupted/terminated
 	cleanupCommands = append(cleanupCommands, func() {
-		logger.Info("stopping listening for blocks...")
-		blockHeaderForTSS.Close()
-		logger.Info("block listener stopped")
 		logger.Info("stop listening for events...")
 		cancelEventCtx()
 		<-eventBus.Done()
 		logger.Info("event listener stopped")
+		logger.Info("stopping subscribers...")
+		for _, subscription := range subscriptions {
+			<-subscription.Done()
+		}
```

Comment:
this change is independent of the dependency update, but the cleanup should wait for _all_ subscriptions to complete

Reply:
There was an issue noticed by a validator that `vald` would not stop with a SIGTERM signal. It seemed like vald never cleaned up properly? #1016